src/spatialdata/_core/query/spatial_query.py

```diff
@@ -102,6 +104,165 @@ def _get_bounding_box_corners_in_intrinsic_coordinates(
+
+    inverse = spatial_transform_bb_axes.inverse()
+    assert isinstance(inverse, Affine)
+
+    return transform(polygon_gdf, inverse)
+
+
+def _get_axes_of_tranformation(
+    element: SpatialElement, target_coordinate_system: str
+) -> tuple[ArrayLike, tuple[str, ...], tuple[str, ...]]:
+    """
+    Get the transformation matrix and the transformation's axes (ignoring `c`).
+
+    The transformation is the one from the element's intrinsic coordinate system to the query coordinate space.
+    Note that the axes which specify the query shape are not necessarily the same as the axes that are output of the
+    transformation
+
+    Parameters
+    ----------
+    element
+        SpatialData element to be transformed.
+    target_coordinate_system
+        The target coordinate system for the transformation.
+
+    Returns
+    -------
+    m_without_c
+        The transformation from the element's intrinsic coordinate system to the query coordinate space, without the
+        "c" axis.
+    input_axes_without_c
+        The axes of the element's intrinsic coordinate system, without the "c" axis.
+    output_axes_without_c
+        The axes of the query coordinate system, without the "c" axis.
+
+    """
+    from spatialdata.transformations import get_transformation
+
+    transform_to_query_space = get_transformation(element, to_coordinate_system=target_coordinate_system)
+    assert isinstance(transform_to_query_space, BaseTransformation)
+    m = _get_affine_for_element(element, transform_to_query_space)
+    input_axes_without_c = tuple([ax for ax in m.input_axes if ax != "c"])
+    output_axes_without_c = tuple([ax for ax in m.output_axes if ax != "c"])
+    m_without_c = m.to_affine_matrix(input_axes=input_axes_without_c, output_axes=output_axes_without_c)
+    return m_without_c, input_axes_without_c, output_axes_without_c
+
+
+def _adjust_bounding_box_to_real_axes(
+    axes: tuple[str, ...],
+    min_coordinate: ArrayLike,
+    max_coordinate: ArrayLike,
+    output_axes_without_c: tuple[str, ...],
+) -> tuple[tuple[str, ...], ArrayLike, ArrayLike]:
+    """
+    Adjust the bounding box to the real axes of the transformation.
+
+    The bounding box is defined by the user and it's axes may not coincide with the axes of the transformation.
+    """
+    if set(axes) != set(output_axes_without_c):
+        axes_only_in_bb = set(axes) - set(output_axes_without_c)
+        axes_only_in_output = set(output_axes_without_c) - set(axes)
+
+        # let's remove from the bounding box whose axes that are not in the output axes (e.g. querying 2D points with a
+        # 3D bounding box)
+        indices_to_remove_from_bb = [axes.index(ax) for ax in axes_only_in_bb]
+        axes = tuple([ax for ax in axes if ax not in axes_only_in_bb])
+        min_coordinate = np.delete(min_coordinate, indices_to_remove_from_bb)
+        max_coordinate = np.delete(max_coordinate, indices_to_remove_from_bb)
```

Comment:
`min_coordinate` and `max_coordinate` are with respect to axes right? If so

```python
idx_bb = np.in1d(axes, output_axes_without_c)
idx_out = np.in1d(output_axes_without_c, axes)

min_coordinate = min_coordinate[idx_out]
max_coordinate = max_coordinate[idx_out]
```

Reply:
if yes, I would change `axes` to `axes_bb` and maybe `output_axes_without_c ` to `axes_out_without_x` to anyway make them consistent

Reply:
also, do you need to check (maybe done outside) that min/max coordinate matches in len with some axes (either bb or out)

Reply:
> min_coordinate and max_coordinate are with respect to axes right? If so
>
> idx_bb = np.in1d(axes, output_axes_without_c)  
> idx_out = np.in1d(output_axes_without_c, axes)  

this still returns a boolean mask, so it's basically equivalent to what I wrote; I'll keep the old code.

min_coordinate = min_coordinate[idx_out]  
max_coordinate = max_coordinate[idx_out]

Reply:
yes, we check that it matches with bb, and we do it `BoundingBoxRequest.__post_init__()`